### PR TITLE
RN-1019: Dependency Updates

### PR DESCRIPTION
**Pull request by joshua-davis-rose**
* Pulling dependencies out of library. Implementing library will need to include them.
* Also forcing sub-dependency version so we don't end up with conflicting/competing dependencies.
* Picking coming from community library instead of react-native.
* component.refs has been deprecated so the way those are handled needed to be updated.
* Fixed the travis build by bringing the library into modern node standards

Basically the Picker component has been pulled out of React so that was the first problem. The next problem was that dueling dependencies between rmbx_react_native, tcomb-form-native, and tcomb-validation meant certain dependencies need to come from the implementing application, not the library, or else have versions forced through resolutions.

### CODE REVIEW

**Review by CarolynWebster — APPROVED** [review]
Looks legit. I look forward to the funeral dance when this is gone!

File: lib/templates/bootstrap/datepicker.android.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-var React = require("react")
-var { View, Text, TouchableNativeFeedback } = require("react-native")
-var { DateTimePickerAndroid } = require("@react-native-community/datetimepicker")
+var React = require('react')
```
Comment (jonstine) [inline lib/templates/bootstrap/datepicker.android.js:1]: why single quotes all of a sudden?

**Review by jonstine — APPROVED** [review]
Looks fine except for some single quotes that snuck their way in. Any reason for that?

File: lib/templates/bootstrap/datepicker.android.js (hunk 1 above)
Comment (joshua-davis-rose) [reply lib/templates/bootstrap/datepicker.android.js:1]: This is a forked version of `tcomb-form-native` that we made tweaks a long time ago to keep running. This round is another bump to keep things rolling. This library has an eslint rule for single quotes, but eslint wasn't actually running so it wasn't being caught. I changed the few instances of double quotes to single quotes to limit the breadth of the changes.

**Comment by joshua-davis-rose** [comment]
@CarolynWebster Could I get your eyes on this again sometime today since I made quite a few changes since your last review?

File: .travis.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 language: node_js
 node_js:
-  - "0.12"
```
Comment (CarolynWebster) [inline .travis.yml:3]: Wow.

File: lib/templates/bootstrap/textbox.js [hunk 3]
```diff
@@ -69,7 +72,7 @@ function textbox(locals) {
           onChange={locals.onChangeNative}
           placeholder={locals.placeholder}
           style={textboxStyle}
-          value={locals.value}
+          value={locals.value ? locals.value.toString() : ''}
```
Comment (CarolynWebster) [inline lib/templates/bootstrap/textbox.js]: Will this set empty textboxes to an empty string? I think `undefined` would be preferred

**Review by CarolynWebster — APPROVED** [review]
One question/nit - but otherwise I think this is good. We should give mobile forms a good QA once this is on RC.

File: lib/templates/bootstrap/textbox.js (hunk 3 above)
Comment (joshua-davis-rose) [reply lib/templates/bootstrap/textbox.js]: I'll check this. I was getting an error saying the field had to be a string, but that was before I tracked down another issue so it may be resolved without this.
